src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -14,14 +14,15 @@ namespace Microsoft.AspNet.Routing.Template
 {
     public class TemplateBinder
     {
-        private readonly IReadOnlyDictionary<string, object> _defaults;
+        private readonly RouteValueDictionary _defaults;
+        private readonly RouteValueDictionary _filters;
         private readonly RouteTemplate _template;
         private readonly UrlEncoder _urlEncoder;
 
         public TemplateBinder(
             RouteTemplate template,
             UrlEncoder urlEncoder,
-            IReadOnlyDictionary<string, object> defaults)
```

Comment:
Declaring this as `IReadOnlyDictionary<>` was also resulting in a boxed enumerator per operation.